pgml-dashboard/src/lib.rs

```diff
@@ -397,9 +400,13 @@ pub async fn models_index(cluster: Cluster) -> Result<ResponseOk, errors::Error>
 #[get("/models/<id>")]
 pub async fn models_get(cluster: Cluster, id: i64) -> Result<ResponseOk, errors::Error> {
     let model = models::Model::get_by_id(cluster.pool(), id).await?;
-    let snapshot = models::Snapshot::get_by_id(cluster.pool(), model.snapshot_id).await?;
+    let mut snapshot = models::Snapshot::get_by_id(cluster.pool(), model.snapshot_id).await?;
     let project = models::Project::get_by_id(cluster.pool(), model.project_id).await?;
 
+    if !snapshot.relation_exists(cluster.pool()).await? {
+        snapshot.relation_name = String::from("");
```

Comment:
I think an additional boolean attribute representing "deleted" rather than blanking the original relation name might be useful, and then we can additionally present this state in the UX. We can also cache the `relation_exists` logic for use in multiple places this way. 

Expanding on the concept that a materialized snapshot may be deleted if it's actual data is deleted, and a non materialized snapshot is deleted if the table is dropped.

Reply:
Snapshots has a new boolean field `exists` along with `table_size`. They are part of SQL queries now instead of standalone functions.